### PR TITLE
Add Toc extension for RST (restructuredText)

### PR DESCRIPTION
**Pull request by matclab**

### CODE REVIEW

File: autoload/vista/extension/rst.vim [hunk 1]
```diff
@@ -0,0 +1,89 @@
+" Heavily inspired  from https://raw.githubusercontent.com/Shougo/unite-outline/master/autoload/unite/sources/outline/defaults/rst.vim
```
Comment (liuchengxu) [inline autoload/vista/extension/rst.vim]: You can add the header for this new file :).

```vim
" Copyright (c) 2019 Your Name
" MIT License
" vim: ts=2 sw=2 sts=2 et
```

File: autoload/vista/extension/rst.vim [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+function! s:Render(data) abort
+  " {'lnum': 1, 'level': '4', 'text': 'Vista.vim'}
+  let data = a:data
+
+  let rows = []
+
+  for line in data
+    let level = line.level
+    let text = vista#util#Trim(line['text'])
+    let lnum = line.lnum
```
Comment (liuchengxu) [inline autoload/vista/extension/rst.vim]:
```suggestion
   " line.lnum` is 0-based, but the lnum of vim is 1-based.
    let lnum = line.lnum + 1
```

File: autoload/vista/extension/rst.vim [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        call add(headers, l:item)
+        let idx += 1
+    endif
+    let idx += 1
+ endwhile
+
+  return headers
+endfunction
+
+let s:default_icon = get(g:, 'vista_icon_indent', ['╰─▸ ', '├─▸ '])
```
Comment (liuchengxu) [inline autoload/vista/extension/rst.vim]: Move the consts on the top if possisble.

File: autoload/vista/extension/rst.vim (hunk 1 above)
Comment (matclab) [reply autoload/vista/extension/rst.vim]: done in 83fcdbe

File: autoload/vista/extension/rst.vim (hunk 3 above)
Comment (matclab) [reply autoload/vista/extension/rst.vim]: done in 1c84ad7

File: autoload/vista/extension/rst.vim (hunk 2 above)
Comment (matclab) [reply autoload/vista/extension/rst.vim]: done in fed3cac.

Thanks for the review.

File: autoload/vista/extension/rst.vim [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+    " Check the matching strictly.
+    if matched_line =~# '^\([[:punct:]]\)\1\{3,}$' && line !~# '^\s*$'
+        if idx > 1 && lines[idx - 1] == matched_line
+          " Title
+          let item = {'lnum': idx, 'text': l:line, 'level': 1}
+        else
+          " Sections
+          let item = {'lnum': idx, 'text': l:line}
+          let adchar = matched_line[0]
+          if !has_key(l:adornment_levels, adchar)
```
Comment (liuchengxu) [inline autoload/vista/extension/rst.vim]:
```suggestion
          if !has_key(l:adornment_levels, adchar)
            let l:adornment_levels[adchar] = l:adornment_id
            let l:adornment_id += 1
          endif
```

**Comment by matclab** [comment]
I've just seen that per #58, I should be able to use [rst2ctags](https://github.com/jszakmeister/rst2ctags) to produce the nested view. However simply adding the command to `g:vista_ctags_cmd` does not work as I guess some file shall also be added in `autoload/vista/types/uctags/`. Is this documented somewhere ?

**Comment by liuchengxu** [comment]
It's not well documented. Actually the tree view of ctags is not very reliable for the design flaw, see https://github.com/liuchengxu/vista.vim/issues/123#issuecomment-552620940 . An overhaul for the ctags's tree view is needed, but I don't know when it will be done. We could use this for now, at least it works, although there is a room to improve, I'll see if there is something I can do later.

**Comment by liuchengxu** [comment]
Merged, thanks!

**Comment by matclab** [comment]
You're welcome. Thank you for making Vista ! ;-)

**Pull request merged**